specification/nginx/resource-manager/NGINX.NGINXPLUS/preview/2025-03-01-preview/swagger.json

```diff
@@ -0,0 +1,2850 @@
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v6/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/DeploymentNameParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v6/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successfully listed all deployment's custom Waf policies",
+            "schema": {
+              "$ref": "#/definitions/NginxDeploymentWafPolicyListResponse"
+            }
+          },
+          "default": {
+            "description": "Default error response.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v6/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "WafPolicy_List": {
+            "$ref": "./examples/WafPolicy_List.json"
+          }
+        }
+      }
+    },
+    "/providers/Nginx.NginxPlus/operations": {
+      "get": {
+        "tags": [
+          "NginxDeployment"
+        ],
+        "summary": "List all operations provided by Nginx.NginxPlus for the 2024-11-01-preview api version.",
+        "operationId": "Operations_List",
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v6/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v6/types.json#/definitions/OperationListResult"
+            }
+          },
+          "default": {
+            "description": "Default error response.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v6/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "Operations_List": {
+            "$ref": "./examples/Operations_List.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "ScaleProfile": {
+      "type": "object",
+      "description": "The autoscale profile.",
+      "required": [
+        "name",
+        "capacity"
+      ],
+      "properties": {
+        "name": {
+          "type": "string",
+          "example": "ExampleProfile"
+        },
+        "capacity": {
+          "type": "object",
+          "required": [
+            "min",
+            "max"
+          ],
+          "description": "The capacity parameters of the profile.",
+          "x-ms-client-flatten": true,
```

Comment:
>    "x-ms-client-flatten": true, [](http://example.com/codeflow?start=6&length=32)

Moving flatten here seems like it will change the shape of the SDK generated object in a breaking way, is this intended as a bugfix??

Reply:
We've had this in all our previous API versions and no SDK has been broken